plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -0,0 +1,403 @@
+        else {
+            if (originalFieldName.contains(DOT_SYMBOL)) {
+
+                Map<String, Object> sourceObject = ingestDocument.getSourceAndMetadata();
+
+                ArrayList<Object> fieldValueList = JsonPath
+                    .using(suppressExceptionConfiguration)
+                    .parse(sourceObject)
+                    .read(originalFieldName);
+
+                String originalFieldValueAsString = toString(fieldValueList);
+                updateModelParameters(modelInputFieldName, originalFieldValueAsString, modelParameters);
+            }
+        }
+    }
+
+    /**
+     * support multiple document fields to map to the same model input fields,
+     * check if the key existed, then append to existed value to be a list.
+     */
+    private void updateModelParameters(String modelInputFieldName, String originalFieldValueAsString, Map<String, String> modelParameters) {
+
+        if (modelParameters.containsKey(modelInputFieldName)) {
+            Object existingValue = modelParameters.get(modelInputFieldName);
+            List<Object> updatedList = (List<Object>) existingValue;
+            updatedList.add(originalFieldValueAsString);
+            modelParameters.put(modelInputFieldName, toString(updatedList));
+        } else {
+            modelParameters.put(modelInputFieldName, originalFieldValueAsString);
+        }
+
+    }
+
+    private String getFieldPath(IngestDocument ingestDocument, String originalFieldName) {
+        final boolean fieldPathIsNullOrEmpty = Strings.isNullOrEmpty(originalFieldName);
+
+        if (fieldPathIsNullOrEmpty) {
+            return null;
+        }
+        final boolean hasFieldPath = ingestDocument.hasField(originalFieldName, true);
+        if (!hasFieldPath) {
+            return null;
+        }
+        return originalFieldName;
+    }
+
+    /**
+     * appendFieldValue to the ingestDocument without changing the source
+     */
+
+    private void appendFieldValue(
+        ModelTensorOutput modelTensorOutput,
+        String originalModelOutputFieldName,
+        String newDocumentFieldName,
+        IngestDocument ingestDocument
+    ) {
+        Object modelOutputValue = null;
+
+        if (modelTensorOutput.getMlModelOutputs() != null && modelTensorOutput.getMlModelOutputs().size() > 0) {
+
+            modelOutputValue = getModelOutputValue(modelTensorOutput, originalModelOutputFieldName, ignoreMissing);
+
+            List<String> dotPathsInArray = writeNewDotPathForNestedObject(ingestDocument.getSourceAndMetadata(), newDocumentFieldName);
+
+            if (dotPathsInArray.size() == 1) {
+                ValueSource ingestValue = ValueSource.wrap(modelOutputValue, scriptService);
+                TemplateScript.Factory ingestField = ConfigurationUtils
+                    .compileTemplate(TYPE, tag, newDocumentFieldName, newDocumentFieldName, scriptService);
+                ingestDocument.appendFieldValue(ingestField, ingestValue, false);
```

Comment:
Why use `appendFieldValue` ? I think we should use `ingestDocument.setFieldValue(ingestField, ingestValue);`

To avoid overriding user's existing field, we should check if `ingestField` already exists.

Reply:
This is a good question, 

if the ingestField already exist, the appendFieldValue will append the new value after the existing value. 

If changing to setFieldValue, we can enable allowDuplicates is false, then it will check for field already exist, if field existed, throw out exception. 

will change to setFieldValue and do not allow duplication field and do not allow append to the existed field.

Reply:
changed to setFieldValue and added existed field check

Reply:
https://github.com/opensearch-project/ml-commons/blob/84abae78369b943e634cc22039667ef24c173939/plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java#L199

https://github.com/opensearch-project/ml-commons/blob/84abae78369b943e634cc22039667ef24c173939/plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java#L325